delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+            pool_list = self.get_pool(None)
+            for pool in pool_list:
+                total_capacity += pool['total_capacity']
+                used_capacity += pool['used_capacity']
+                free_capacity += pool['free_capacity']
+            storage_model = {
+                "name": storage_map['ClusterName'],
+                "vendor": constant.STORAGE_VENDOR,
+                "model": model_map.get('Model'),
+                "status": status,
+                "serial_number": serial_number.replace(')', ''),
+                "firmware_version": version[0],
+                "location": location_map['Location'],
+                "total_capacity": total_capacity,
+                "raw_capacity": raw_capacity,
+                "used_capacity": used_capacity,
+                "free_capacity": free_capacity
+            }
+            return storage_model
+
+    def get_storage(self):
+        try:
+            storage_info = self.ssh_do_exec([constant.STORAGE_INFO_COMMAND])
+            model_info = self.ssh_do_exec([constant.STORAGE_MODEL_COMMAND])
+            location_info = self.ssh_do_exec(([constant.LOCATION_COMMAND]))
+            storage_map_list = []
+            model_map_list = []
+            version_map_list = []
+            location_map_list = []
+            serial_map_list = []
+            self.split_value_map_list(
+                model_info, model_map_list, 'Model', split=":")
+            self.split_value_map_list(
+                storage_info, storage_map_list, 'ClusterName', split="=")
+            self.split_value_map_list(
+                model_info, version_map_list, 'Software', split=":")
+            self.split_value_map_list(
+                location_info, location_map_list, 'Location', split=':')
+            self.split_value_map_list(
+                model_info, serial_map_list, 'Hardware', split=':')
+            storage_model = \
+                self.format_storage_info(
+                    storage_map_list, model_map_list, version_map_list,
+                    location_map_list, serial_map_list)
+            return storage_model
+        except exception.DelfinException as e:
+            err_msg = "Failed to get storage from " \
+                      "hitachi nas: %s" % (six.text_type(e.msg))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get storage from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_disk(self, storage_id):
+        try:
+            disk_info = self.ssh_do_exec([constant.DISK_INFO_COMMAND])
+            disk_map_list = []
+            self.split_value_map_list(
+                disk_info, disk_map_list, 'Capacity', split=":")
+            disks_list = []
+            for disk_map in disk_map_list:
+                if 'Status' in disk_map:
```

Comment:
Why is this `if` sentence is needed?

Reply:
Because parsed data may be dirty data with incomplete or useless information in the data object map, only if this key exists in the data object map can it be confirmed that it has completed reliable data, and then proceed to the next step of parsi